[limrik] iP

limrik opened the pull request:
## Project Duke as Uncle Bob
- **easy** to use
- text based

All you need to do is 
1. Download it from [here](https://github.com/limrik/ip/releases/tag/A-Jar)
2. Let _Uncle Bob_ manage your tasks for you

Features
- [x] managing todos, deadlines and events
- [ ] reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` Java method:
``` Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
> Code is like humor. When you have to explain it, it’s bad. – Cory House :100:

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

limrik committed:
- [x] renamed chatbot
- [x] implemented greet and exit functions

limrik committed:
feat: rename bot, create greet and exit functions

limrik committed:
feat: echo user commands

limrik committed:
feat: store and display user input

limrik committed:
feat: add ability to mark and undone tasks

limrik committed:
feat: support tracking for different types of tasks

limrik committed:
feat: basic tests

limrik committed:
refactor: improve readability of switch statements

limrik committed:
feat: handle errors

limrik committed:
feat: implemented delete function

limrik committed:
feat: load and save from hard disk

limrik committed:
feat: allow event and deadline to handle dates

limrik committed:
Merge branch 'Level-8'

limrik committed:
Implement more OOP

limrik committed:
Merge tag 'A-MoreOOP'

limrik committed:
Merge remote-tracking branch 'origin/add-gradle-support'

limrik committed:
Add dummy test

limrik committed:
Refactor code to display correct output

limrik committed:
Add tests for DeadlineCommand and TodoCommand

limrik committed:
Merge branch 'A-JUnit'

limrik committed:
Add JavaDoc comments

limrik committed:
Refactor code to adhere to java coding standards

limrik committed:
Implement find functionality

limrik committed:
Merge branch 'Level-9'

limrik committed:
Merge branch 'A-CodingStandard'

limrik committed:
Resolve merge conflicts

limrik committed:
Resolved checkstyle errors

limrik committed:
Bug fix

limrik committed:
Merge branch 'A-CheckStyle'

limrik committed:
Setup basic GUI

limrik committed:
Setup basic GUI

guohuang88 commented on src/main/java/commands/ExitCommand.java:
```diff
@@ -0,0 +1,23 @@
+    public static final String COMMAND_WORD = "bye";
+    private static final String EXIT_MESSAGE = "Bye! Hope to see you again soon!";
+
+    /**
+     * Executes the ExitCommand, exiting the application
+     * @param tasks   The TaskList representing the collection of tasks.
+     * @param storage The Storage object handling storage operations.
+     * @return
+     */
```
```suggestion
    /**
     * Executes the ExitCommand, exiting the application.
     * @param tasks   The TaskList representing the collection of tasks.
     * @param storage The Storage object handling storage operations.
     * @return Exit message.
     */
```
Perhaps adding a description for the return in the JavaDoc would be better?

guohuang88 commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,98 @@
+        this.by = by;
+    }
+
+    /**
+     * Constructs a Task with the specified symbol, description, start date, and end date.
+     *
+     * @param symbol      the symbol representing the type of the task
+     * @param description the description of the task
+     * @param start       the start date for events with a specified duration
+     * @param end         the end date for events with a specified duration
+     */
```
```suggestion
    /**
     * Constructs a Task with the specified symbol, description, start date, and end date.
     *
     * @param symbol      The symbol representing the type of the task.
     * @param description The description of the task.
     * @param start       The start date for events with a specified duration.
     * @param end         The end date for events with a specified duration.
     */
```
I really like how you start the description for the params at the same starting point as it looks very neat!
Perhaps it would be better if you start each description with a capital letter and ends of with a full stop?

guohuang88 commented on src/test/java/storage/StorageStub.java:
```diff
@@ -0,0 +1,55 @@
+ */
+public class StorageStub extends Storage {
+
+    private boolean simulateIoException;
+    private boolean appendToFileCalled;
```
Perhaps there is a more intuitive way to name this boolean variable here?

guohuang88 commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,51 @@
+
+    public TaskList() {}
+
+    public void addTasks(Task task) {
```
```suggestion
    public void addTask(Task task) {
```
Perhaps it will be better if this method name is in singular form since each time you are only adding one task?

guohuang88 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,57 @@
+        String[] inputs = userInput.split(" ");
+        String command = inputs[0];
+        String message = String.join(" ", Arrays.copyOfRange(inputs, 1, inputs.length));
+        switch (command) {
+        case ExitCommand.COMMAND_WORD:
+            return new ExitCommand();
+        case ListCommand.COMMAND_WORD:
+            return new ListCommand();
+        case MarkCommand.COMMAND_WORD:
+            return new MarkCommand(message);
+        case UnmarkCommand.COMMAND_WORD:
+            return new UnmarkCommand(message);
+        case TodoCommand.COMMAND_WORD:
+            return new TodoCommand(message);
+        case DeadlineCommand.COMMAND_WORD:
+            return new DeadlineCommand(message);
+        case EventCommand.COMMAND_WORD:
+            return new EventCommand(message);
+        case DeleteCommand.COMMAND_WORD:
+            return new DeleteCommand(message);
+        case FindCommand.COMMAND_WORD:
+            return new FindCommand(message);
+        default:
+            return new ExitCommand();
```
I like how you use switch here as it looks very neat and easy to understand!

guohuang88 commented on src/main/java/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,42 @@
+    public String execute(TaskList tasks, Storage storage) {
+        try {
+            int index = Integer.parseInt(message);
+            Task removed = tasks.get(index - 1);
```
```suggestion
            Task removedTask = tasks.get(index - 1);
```
Perhaps a more intuitive variable name here?

guohuang88 reviewed (commented):
Your code looks really neat!
Some Java coding standard changes requested, you may take a look!

Murugan-Maniish commented on src/main/java/commands/ExitCommand.java:
```diff
@@ -0,0 +1,23 @@
+
+import storage.Storage;
+import task.TaskList;
+/**
+ * Represents the command used to exit the application.
+ */
```
Having a empty line at the start after import statements might look nicer

Murugan-Maniish commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,51 @@
+/**
+ * * Represents a collection of tasks in the Duke application.
+ */
+public class TaskList implements Iterable<Task> {
```
I like how you implemented this interface for TaskList

Murugan-Maniish commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,98 @@
+    protected String symbol;
+    protected String description;
+    protected boolean isDone;
+    protected LocalDate by;
+    protected LocalDate start;
+    protected LocalDate end;
```
Any reason why these variables were put under Task as opposed to the alternative in the Deadline and Event classes?

Murugan-Maniish commented on src/main/java/commands/TodoCommand.java:
```diff
@@ -0,0 +1,53 @@
+ */
+public class TodoCommand extends Command {
+
+    public static final String COMMAND_WORD = "todo";
```
Any reason for adding this constant to each command class rather than using the string directly in Parser?

Murugan-Maniish reviewed (commented):
LGTM. Naming of classes, variables and methods were all well done.

limrik committed:
Add assertions to mark and unmark commands

Mark command asserts that task status is
updated to 1

Unmark command asserts that task status
is updated to 0

limrik committed:
Revert "Add assertions to mark and unmark commands"

This reverts commit 20050ff3c6ccf6eba5e075de841c8b27ae0db281.

limrik committed:
Add assertions to mark and unmark commands

Mark command asserts that task status is
updated to 1

Unmark command asserts that task status
is updated to 0

limrik committed:
Merge pull request #1 from limrik/A-Assertions

A assertions

limrik committed:
Refactor code to improve code quality

Made the happy path more prominent

Removed nested catch statements

limrik committed:
Merge pull request #2 from limrik/A-CodeQuality

Refactor code to improve code quality

limrik committed:
Setup Github CI

limrik committed:
Fix checkstyle errors

limrik committed:
Removed unnecessary test files

limrik committed:
Implement remind command

Display tasks that are due on the current
date when the remind command is called

Display no task found if there are no tasks
due on the current date

limrik committed:
Merge pull request #3 from limrik/BCD-Extension

Implement remind command

limrik committed:
Add more tests

limrik committed:
Merge pull request #4 from limrik/A-MoreTesting

Add more tests

limrik committed:
Rename to UncleBob

limrik committed:
Improve GUI

limrik committed:
Remove unused imports

limrik committed:
Merge pull request #5 from limrik/A-BetterGui

Improve GUI

limrik committed:
Add representative screenshot and name in title bar

limrik committed:
Refactor files

limrik committed:
Update user guide

limrik committed:
Merge pull request #7 from limrik/A-UserGuide

Update user guide

limrik committed:
Update readme

limrik committed:
Update readme

limrik committed:
Fix build issues

limrik committed:
Refactor files

limrik committed:
Replace Ui.png

limrik committed:
Update readme

limrik committed:
Update readme

limrik committed:
Change main class name
